src/root.zig

```diff
@@ -210,6 +210,15 @@ pub fn AlgorithmType(
             self.max_needle = max_needle;
         }
 
+        // Check is there is enough memory allocated
+        pub fn hasSize(self: *Self, max_haystack: usize, max_needle: usize) bool {
+            if (self.max_haystack < max_haystack or self.max_needle < max_needle) {
+                return false;
+            } else {
+                return true;
+            }
```

Comment:
```suggestion
            return (max_haystack <= self.max_haystack) and (max_needle <= self.max_needle);
```
It's easier just to check the positive case.